src/main/java/org/mediawiki/sparql/mwontop/http/SPARQLActions.java

```diff
@@ -134,7 +134,26 @@ private Response evaluateTupleQuery(TupleQuery query, Request request) {
         return Response.ok(
                 (StreamingOutput) outputStream -> {
                     try {
-                        query.evaluate(format.getService().getWriter(outputStream));
+                        TupleQueryResultWriter writer = format.getService().getWriter(outputStream);
+                        TupleQueryResult result = query.evaluate();
+
+                        writer.startQueryResult(result.getBindingNames());
```

Comment:
Maybe move the conversion logic in its own function that takes and return a BindingSet?